File: littlepony/src/app/race.service.ts
import { Injectable } from '@angular/core';
import { Race } from './race';
import { RACES } from './race-mock';
import { of, Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class RaceService {

  races: Array<Race>;
  constructor() { 
    this.races = [];
    this.races = RACES;
  }

  getAllRaces(): Observable<Array<Race>>
  {
    return of(this.races);
  }

  addRace(r: Race)
  {
      this.races.push(r);
  }
}

File: TPs JAVA et Angular/AngularEtSpring/src/main/java/fr/dta/AngularEtSpring/service/RaceService.java
package fr.dta.AngularEtSpring.service;

import java.util.ArrayList;
import java.util.List;

import fr.dta.AngularEtSpring.model.Race;

public class RaceService
{

	private List<Race> races;

	public RaceService()
	{
		this.races = new ArrayList<Race>();
	}

	public List<Race> getAllRaces()
	{
		return this.races;
	}

	public void addRace(Race r)
	{
		this.races.add(r);
	}
}
